python/CMakeLists.txt

```diff
@@ -0,0 +1,41 @@
+# Set up pybind11 as an external project.
+set(pb11_src_dir "${CMAKE_CURRENT_SOURCE_DIR}/pybind11")
+
+find_package(Git)
+if(NOT EXISTS "${pb11_src_dir}/.git")
+    set(git_failed)
+
+    if(GIT_FOUND)
+        message(STATUS "Updating the Pybind11 submodule ${pb11_src_dir}")
+        execute_process(
+            COMMAND "${GIT_EXECUTABLE}" submodule update --init "${pb11_src_dir}"
+            WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}"
+            ERROR_VARIABLE git_error
+            RESULT_VARIABLE git_result)
+        if(NOT git_result EQUAL 0)
+            set(git_failed "${git_error}")
+        endif()
+    else()
+        set(git_failed "git not found")
+    endif()
+
+    if(git_failed)
+        message(WARNING "Unable to update the Pybind11 submodule: ${git_failed}")
+    endif()
+
+endif()
+
+# Set up pybind11, which is used to generate python bindings.
+# We can just add the pybind11 directory, instead of find_package().
+set(PYBIND11_CPP_STANDARD -std=c++11)
+add_subdirectory(pybind11)
+
+pybind11_add_module(pyarb arb.cpp)
+
+target_link_libraries(pyarb LINK_PUBLIC ${ARB_LIBRARIES})
+target_link_libraries(pyarb LINK_PUBLIC ${EXTERNAL_LIBRARIES})
+
```

Comment:
(This relates also to the use of the global setting of `CMAKE_POSITION_INDEPENDENT_CODE` in the top level CMakeLists.txt.)

We should decide to either have PIC for all libraries, independent of python support, or build the static libraries without it and the dynamic ones with. Getting CMake to do the latter is tedious but possible.

Extra care, ultimately, will have to be taken to use a shared TBB library for `pyarb` if we allow a static TBB library for `libarbor.a`.

Reply:
I had to use PIC for the static libraries, otherwise it isn't possible to link the `libarbor.a` into the python module.

Why would we want to use PIC everywhere all the time, instead of just when building Python?

Reply:
(Talked about this offline, but commenting for the record.) The issue is one of consistency: building with or without Python shouldn't change the build artefacts, i.e. libarbor.a